test/rules/prefer_mixin.dart

```diff
@@ -4,8 +4,103 @@
 
 // test w/ `pub run test -N prefer_mixin`
 
+import 'dart:collection';
+
+import 'dart:convert';
+
 class A {}
+
 class B extends Object with A {} // LINT
 
 mixin M {}
+
 class C with M {} // OK
+
+class I with IterableMixin //OK
```

Comment:
I don't have strong feelings about it, but you could significantly shorten the test code by writing `abstract class I with IterableMixin {} // OK`. If it's abstract it doesn't need to implement the abstract methods from the mixin.